### PR TITLE
gstreamer-good: support cross-compilation

### PR DESCRIPTION
**Pull request by Mindavi**
###### Description of changes

Add support for cross-compilation. Also ensures pkg-config file is generated by ilbshout.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux (cross-compiled)
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @mehmooda which is hacking on gstreamer and cross as well (but then for Windows)

### CODE REVIEW

File: pkgs/development/libraries/gstreamer/good/default.nix [hunk 1]
```diff
@@ -57,13 +58,20 @@ stdenv.mkDerivation rec {
...
+  depsBuildBuild = [ pkg-config ];
+
   nativeBuildInputs = [
     pkg-config
     python3
     meson
     ninja
     gettext
     nasm
+    orc
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gstreamer/good/default.nix:72]: Can this be removed from buildInputs?

**Review by SuperSandro2000 — COMMENTED** [review]
otherwise LGTM

**Comment by Mindavi** [comment]
Validated that dropping orc from buildInputs doesn't change the output, checked with diffoscope.

File: pkgs/development/libraries/gstreamer/good/default.nix (hunk 1 above)
Comment (Mindavi) [reply pkgs/development/libraries/gstreamer/good/default.nix:72]: Dropped. Thanks!

**Comment by mehmooda** [comment]
Anyway after a long detour trying to compile something that isn't this. I'm sure orc is meant to be both a buildInput and nativeBuildInput. orcc is native and liborc as a buildinput

**Comment by Mindavi** [comment]
Is liborc actually used in gst-plugins-good then? I don't see any reference to it in the diff from diffoscope. It seems like liborc is a JIT compiler of sorts..?

**Comment by symphorien** [comment]
this caused a build failure in nheko

**Pull request merged**